cdk/lib/payment-api.ts

```diff
@@ -1,22 +1,409 @@
+      description: "ARN of the kinesis stream to write events to",
+    });
+
+    const contributionsStoreSqsQueueCodeArn = new GuStringParameter(
+      this,
+      "ContributionsStoreSqsQueueCodeArn",
+      {
+        description:
+          "For the PROD stack you still need to supply this, because PROD instances need access to both PROD and CODE contribution store queues.",
+      }
+    );
+
+    const contributionsStoreSqsQueueProdArn = new GuStringParameter(
+      this,
+      "ContributionsStoreSqsQueueProdArn",
+      {
+        description:
+          "For the CODE stack you can leave this empty since it won't be used.  For the PROD stack you need to set it.",
+      }
+    );
+
+    const sqsKmsArn = new GuStringParameter(this, "SqsKmsArn", {
+      description: "ARN of the KMS key for encrypting SQS data",
+    });
+
     const yamlTemplateFilePath = join(
       __dirname,
       "../..",
       "support-payment-api/src/main/resources/cloud-formation.yaml"
     );
+
     new CfnInclude(this, "YamlTemplate", {
       templateFile: yamlTemplateFilePath,
       parameters: {
         Stage: props.stage,
+        EmailSqsQueueCodeArn: emailSqsCodeArn,
+        EmailSqsQueueProdArn: emailSqsProdArn,
+        OphanRole: ophanRole,
+        KinesisStreamArn: kinesisStreamArn,
+        ContributionsStoreSqsQueueCodeArn: contributionsStoreSqsQueueCodeArn,
+        ContributionsStoreSqsQueueProdArn: contributionsStoreSqsQueueProdArn,
+        SqsKmsArn: sqsKmsArn,
+      },
+    });
+
+    // TODO: Should these remain as cloudformation parameters?
+    const projectName = "payment-api";
+    const projectVersion = "0.1";
+
+    const app = "payment-api";
+    const userData = `#!/bin/bash -ev
+    mkdir /etc/gu
+
+    cat > /etc/gu/stage <<'EOF'
+    ${this.stage}
+    EOF
+
+    aws --region ${this.region} s3 cp s3://membership-dist/${this.stack}/${this.stage}/${app}/${projectName}_${projectVersion}_all.deb /tmp
+    dpkg -i /tmp/${projectName}_${projectVersion}_all.deb
+    /opt/cloudwatch-logs/configure-logs application ${this.stack} ${this.stage} ${app} /var/log/${app}/application.log`;
+
+    const playApp = new GuPlayApp(this, {
+      app: "payment-api",
+      access: { scope: AccessScope.PUBLIC },
+      certificateProps: {
+        domainName: props.domainName,
+        hostedZoneId: "Z1E4V12LQGXFEC",
       },
+      monitoringConfiguration: { noMonitoring: true },
+      instanceType: InstanceType.of(InstanceClass.T4G, InstanceSize.SMALL),
+      scaling: props.scaling,
+      userData,
+      roleConfiguration: {
+        additionalPolicies: [
+          new GuAllowPolicy(this, "CloudwatchLogs", {
+            actions: [
+              "logs:CreateLogGroup",
+              "logs:CreateLogStream",
+              "logs:PutLogEvents",
+            ],
+            // TODO: Should we use a more specific resource here?
+            resources: ["arn:aws:logs:*:*:*"],
```

Comment:
The current template actually creates a specific log group and then just gives permisions for that specific log group. We could do that in CDK but that would require creating a new log group which I'm not sure we want to do. For now, we've just given access to all log groups